src/v0/destinations/mailchimp/utils.js

```diff
@@ -186,9 +188,14 @@ const checkIfDoubleOptIn = async (apiKey, datacenterId, audienceId) => {
       }
     });
   } catch (error) {
-    throw new CustomError(
-      "[Mailchimp]:: User does not have access to the requested operation",
-      error.status || 400
+    const status = error.status || 400;
+    throw new NetworkError(
+      "User does not have access to the requested operation",
+      status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status)
+      },
+      error
```

Comment:
Don't send `error` object itself here. Send the appropriate response object, if any.